mock-url.mjs

```diff
@@ -1,98 +1,129 @@
+	flex-grow: 1;
+	flex-direction: row;
 }
 mock-url .url .base {
-    flex-grow: 0;
-    flex-shrink: 0;
-    line-height: 24px;
-    padding: 1px 0px;
+	flex-grow: 0;
+	flex-shrink: 0;
+	line-height: 24px;
+	padding: 1px 0px;
 }
 mock-url .input {
-    flex-grow: 1;
+	flex-grow: 1;
 }
 mock-url .back, mock-url .forward, mock-url .reload {
-    font-size: 20px;
-    font-family: Verdana,sans-serif;
-    border: solid 1px black;
-    padding: 5px;
-    line-height: 20px;
-    flex-grow: 0;
-    cursor: pointer;
+	font-size: 20px;
+	font-family: Verdana,sans-serif;
+	border: solid 1px black;
+	padding: 5px;
+	line-height: 20px;
+	flex-grow: 0;
+	cursor: pointer;
 }
 mock-url .back:hover, mock-url .forward:hover, mock-url .reload:hover, mock-url input:hover {
-    background-color: #ADD8E6;
+	background-color: #ADD8E6;
 }
 mock-url input {
-    border: none;
-    font-size: 20px;
-    background-color: #efefef;
-    width: 100%;
+	border: none;
+	font-size: 20px;
+	background-color: #efefef;
+	width: 100%;
 }
 
 `;
 
 document.body.appendChild(style);
 
 Component.extend({
-    tag: "mock-url",
-    view: `
-        <div class='location'>
-            <span class='back' on:click='back()'>&#x21E6;</span>
-            <span class='forward' on:click='forward()'>&#x21E8;</span>
-            <span class='reload' on:click='reload()'>&#8635;</span>
-            <div class="url"><span class='base'>URL: {{page}}</span><span class='input'><input value:bind="url" placeholder="#! The hash is empty"/></span></div>
-        </div>`,
+	tag: "mock-url",
+	view: `
+		<div class='location'>
+			<span class='back' on:click='back()'>&#x21E6;</span>
+			<span class='forward' on:click='forward()'>&#x21E8;</span>
+			<span class='reload' on:click='reload()'>&#8635;</span>
+			<div class="url">
+				<span class='base'>URL:{{page}}{{path}}</span>
+				<span class='input'><input value:bind="url" placeholder="#! The hash is empty"/></span>
+			</div>
+		</div>
+	`,
+	ViewModel: DefineMap.extend("MockUrl",{
+		page: {
+			default: "/my-app.html"
+		},
+		pushstate: {
+			default: false
+		},
+		path: {
+			value(prop) {
+				// if pushstate's value isn't changed exit function resolving null.
+				if (!this.pushstate) {
+					prop.resolve(null);
+					return;
+				}
 
-    ViewModel: DefineMap.extend("MockUrl",{
-        page: {
-            default: "/my-app.html"
-        },
-        url: {
-            value(prop) {
-                // When the property is set, update the hash.
-                prop.listenTo(prop.lastSet, function(newVal){
-                    window.location.hash = newVal;
-                });
+				var pushState = window.history.pushState;
+				
+				// prevent history.pushState from refreshing the page
+				window.history.pushState = function(){
+					window.history.replaceState.apply(this, arguments);
+					prop.resolve(location.pathname);
+				};
+				
+				// creating default pathname
+				window.history.replaceState( null, null, '/' );
```

Comment:
If using a `root` other than `/`, won't this cause the URL to navigate away from the app?

Reply:
It'll only navigate if the `location.pathname` is directly edited, as far as I've been able to find.